controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1855,6 +1859,27 @@ func (r *BareMetalHostReconciler) createHostFirmwareSettings(info *reconcileInfo
 	return nil
 }
 
+func (r *BareMetalHostReconciler) hostUpdatePolicySetOwnerReference(info *reconcileInfo) error {
+	// NOTE(janders) the goal here is to ensure that the controller reads the hup resource and adds OwnerReference to it
+	hup := &metal3api.HostUpdatePolicy{}
+	if err := r.Get(info.ctx, info.request.NamespacedName, hup); err != nil {
+		// Error reading the object
```

Comment:
Hmm, forgot, sorry: HTTP not found should be handled differently (i.e. ignored). See the other dependant resources for inspiration.

Reply:
Thank you Dmitry, very helpful pointer. I think I got it now, let's see what CI thinks (local tests passed).